test/powershell/Modules/Microsoft.PowerShell.Management/Set-Location.Tests.ps1

```diff
@@ -231,3 +231,58 @@ Describe "Set-Location" -Tags "CI" {
         }
     }
 }
+
+Describe "Set-Location: Name with special/wildcards characters" -Tags "CI" {
+
+    BeforeAll {
+
+        $startLocation = (Get-Location).Path
+
+        # printable ascii characters excluding alphanumerics
+        $specialChars =  ' !"#$%&''()*+,-./:;<=>?@[\]^_`{|}~	'.ToCharArray()
+        $testCases = @()
+        ForEach ($specialChar in $specialChars) {
+            If ($specialChar -ne '.') { $testCases += @{Name=([string] $specialChar)} }
+            $testCases += @{Name=('foo' + $specialChar)}
+            $testCases += @{Name=($specialChar + 'foo')}
```

Comment:
Please add negative cases:

1. Where the folder has a special character but doesn't match the input with a special character
1. Where the folder has a special character and the input doesn't have a special character.

Reply:
I guess 1 would not be a negative case.   I should match.

Reply:
So for #2, you're thinking a test where something where 'a???b' exists literally on the filesystem and we make sure that a lookup on 'a111b' doesn't match it?

Reply:
Tests added.